### PR TITLE
Register default allocator for host memory

### PR DESCRIPTION
**Pull request by revans2**
This gets us to a point where we can have CUDF java allocating host memory through HostAlloc. But it is not 100% perfect yet. Test pass, but the integration tests appear to have a live lock in them that I need to track down when running with limited memory. It should still be good for us to get started working on other HostMemoryLimit issues.

### CODE REVIEW

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
build

**Review by jbrennan333 — APPROVED** [review]
lgtm

**Review by jbrennan333 — DISMISSED** [review]

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/HostAlloc.scala [hunk 1]
```diff
@@ -163,6 +163,9 @@ private class HostAlloc(nonPinnedLimit: Long) extends HostMemoryAllocator with L
     var allocAttemptFinishedWithoutException = false
     try {
       do {
+        if (retryCount > 0) {
+          logWarning(s"RETRY HOST ALLOC $amount $preferPinned $blocking $retryCount")
```
Comment (abellina) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/HostAlloc.scala:167]: did you want to leave this as warn?

**Comment by revans2** [comment]
build

**Review by jbrennan333 — APPROVED** [review]
Looks good.

**Pull request merged**